errors/TypeSynonymInstance.md

```diff
@@ -3,19 +3,43 @@
 ## Example
 
 ```purescript
-module ShortFailingExample where
+module TypeSynonymInstanceError where
+import Prelude
 
-...
+type Mass = Number
+
+instance showMass :: Show Mass where
+  show mass = "Mass: " <> show mass <> "kg"
 ```
 
 ## Cause
 
-Explain why a user might see this error.
+*Type synonyms* are merely aliases (usually created to avoid verbose type annotations) and do not declare a new unique type. It is still possible to use the synonym and its equivalent type interchangeably:
+
+```purescript
+velocity = 123.456 :: Mass  -- oops
```

Comment:
Velocity is a complex metric, like m/s. Might want to use a simple metric, like meters.

Something like the following, though I think you can take parts of it, as it's relatively long, and maybe not necessary to fully explain.

```
distanceToHome = 123.456 :: Mass
-- Distance is meters, which is one way to use a Number,
--   but it's certainly not a mass. This could easily become a bug.
-- Can fix making it a distinct type, instead of an alias for Number.
```